awpy/analytics/map_control.py

```diff
@@ -0,0 +1,382 @@
+"""Functions for calculating map control values and metrics.
+
+    Example notebook:
+
+    https://github.com/pnxenopoulos/awpy/blob/main/examples/05_Map_Control_Calculations_And_Visualizations.ipynb
+"""
+
+from collections import defaultdict
+
+import numpy as np
+
+from awpy.analytics.nav import area_distance, calculate_tile_area, find_closest_area
+from awpy.data import NAV, NAV_GRAPHS
+from awpy.types import FrameMapControl, GameFrame, GameRound, TeamFrameInfo
+
+
+def approx_neighbors(
+    map_name: str,
+    source_tile_id: int,
+    n_neighbors: int = 5,
+) -> list[tuple[float, int]]:
+    """Approximates neighbors for isolated tiles by finding n closest tiles.
+
+    Args:
+        map_name (str): Map for source_tile_id
+        source_tile_id (int): Area ID as an integer
+        n_neighbors (int): Number of closest tiles/approximated neighbors wanted
+    Returns:
+        List of tuples representing n closest tiles.
+            Tuple[0] -> Distance between source_tile_id and represented tile
+            Tuple[1] -> Area ID of represented tile
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+                    If source_tile_id is not in awpy.data.NAV[map_name]
+                    If the length of point is not 3
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+    if source_tile_id not in NAV[map_name].keys():
+        msg = "Area ID not found."
+        raise ValueError(msg)
+    if n_neighbors < 0:
+        msg = "n_neighbors must be >= 0"
+        raise ValueError(msg)
+
```

Comment:
Yeah that makes sense. My reasoning was that if a user used any of these functions as stand-alone then it would be helpful to have this aggressive checking but some of these functions are private so I can remove the checks for those functions.